tests/test_tasks/test_spatial.py

```diff
@@ -0,0 +1,93 @@
+                "resolution": (10, 10),
+            },
+            (slice(0, 100, None), slice(0, 100, None)),
+        ),
+        (
+            {
+                "bbox": BBox(((729480.0, 4390045.0), (732120.0, 4391255.0)), crs=CRS("32638")),
+                "slice_bbox": BBox(((729480.0, 4390045.0), (732120.0, 4391255.0)), crs=CRS("32638")),
+                "resolution": (10, 10),
+            },
+            (slice(0, 264, None), slice(0, 121, None)),
+        ),
+        (
+            {
+                "bbox": BBox(((729480.0, 4390045.0), (732120.0, 4391255.0)), crs=CRS("32638")),
+                "slice_bbox": BBox(((729480.0, 4390045.0), (732120.0, 4391255.0)), crs=CRS("32638")),
+                "resolution": (10, 10),
+                "limit_x": (50, 100),
+                "limit_y": (100, 200),
+            },
+            (slice(50, 100, None), slice(100, 121, None)),
+        ),
+        (
+            {
+                "bbox": BBox(bbox=[500, 500, 600, 600], crs=CRS.WGS84),
+                "slice_bbox": BBox(bbox=[500, 500, 600, 600], crs=CRS.WGS84),
+                "resolution": (10, 10),
+            },
+            (slice(0, 10, None), slice(0, 10, None)),
+        ),
+        (
+            {
+                "bbox": BBox(((729480.0, 4390045.0), (732120.0, 4391255.0)), crs=CRS("32638")),
+                "slice_bbox": BBox(((729480.0, 4390045.0), (732120.0, 4391255.0)), crs=CRS("32638")),
+                "size": (10, 10),
+            },
+            (slice(0, 10, None), slice(0, 10, None)),
+        ),
+        (
+            {
+                "bbox": BBox(((729480.0, 4390045.0), (732120.0, 4391255.0)), crs=CRS("32638")),
+                "slice_bbox": BBox(((729480.0, 4390045.0), (732120.0, 4391255.0)), crs=CRS("32638")),
+                "size": (10, 10),
+                "limit_x": (50, 100),
+                "limit_y": (100, 200),
+            },
+            (slice(50, 10, None), slice(100, 10, None)),
+        ),
+    ],
+)
+def test_get_array_slices(input_params: Dict[str, Any], expected: Tuple[slice, slice]) -> None:
+    assert get_array_slices(**input_params) == expected
+
+
+@pytest.mark.parametrize(
+    "input_params",
+    [
+        {
+            "bbox": BBox(((729480.0, 4390045.0), (732120.0, 4391255.0)), crs=CRS("32638")),
+            "slice_bbox": BBox(((729480.0, 4390255.0), (730480.0, 4391255.0)), crs=CRS("32638")),
+        },
+        {
+            "bbox": BBox(((729480.0, 4390045.0), (732120.0, 4391255.0)), crs=CRS("32638")),
+            "slice_bbox": BBox(((729480.0, 4390045.0), (732120.0, 4391255.0)), crs=CRS("32638")),
+            "resolution": (10, 10),
+            "size": (10, 10),
+        },
+        {
+            "bbox": BBox(((729480.0, 4390045.0), (732120.0, 4391255.0)), crs=CRS("32638")),
+            "slice_bbox": BBox(((729480.5, 4390045.5), (732120.0, 4391255.0)), crs=CRS("32638")),
+            "resolution": (10, 10),
+        },
```

Comment:
add comments (or even better, a test case name) to these, so that it is clear WHY they should fail

Reply:
Add test ids which add reason why test should fail, add it to working cases as well.

Reply:
Currently it is difficult to see which IDs belong to which cases. Consider using `pytest.param` (see [here](https://docs.pytest.org/en/7.1.x/example/parametrize.html#different-options-for-test-ids)) or an id-parser like we did in tests for `eolearn.core.utils.parsing`.

Reply:
done